modules/scripts/startup-script/examples/install_ansible.sh

```diff
@@ -13,41 +13,183 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+REQ_ANSIBLE_VERSION=2.11
+REQ_ANSIBLE_PIP_VERSION=4.10.0
+REQ_PIP_VERSION=18
+REQ_PYTHON3_VERSION=6
+
 apt_wait() {
-	while fuser /var/lib/dpkg/lock >/dev/null 2>&1; do
+	while fuser /var/lib/dpkg/lock 2>/dev/null 2>&1; do
 		echo "Sleeping for dpkg lock"
 		sleep 3
 	done
-	while fuser /var/lib/apt/lists/lock >/dev/null 2>&1; do
+	while fuser /var/lib/apt/lists/lock 2>/dev/null 2>&1; do
 		echo "Sleeping for apt lists lock"
 		sleep 3
 	done
 	if [ -f /var/log/unattended-upgrades/unattended-upgrades.log ]; then
 		echo "Sleeping until unattended-upgrades finishes"
-		while fuser /var/log/unattended-upgrades/unattended-upgrades.log >/dev/null 2>&1; do
+		while fuser /var/log/unattended-upgrades/unattended-upgrades.log 2>/dev/null 2>&1; do
 			sleep 3
 		done
 	fi
 }
 
-if [ ! -h /usr/bin/ansible-playbook ] || [ ! -f /usr/bin/ansible-playbook ]; then
-	if [ -f /etc/centos-release ] || [ -f /etc/redhat-release ] || [ -f /etc/oracle-release ] || [ -f /etc/system-release ]; then
-		if [ ! -f /bin/pip ]; then
-			curl -Os https://bootstrap.pypa.io/pip/2.7/get-pip.py
-			/usr/bin/python get-pip.py
+# Installs any dependencies needed for python based on the OS
+install_python_deps() {
+	if [ -f /etc/debian_version ] || grep -qi ubuntu /etc/lsb-release 2>/dev/null ||
+		grep -qi ubuntu /etc/os-release 2>/dev/null; then
+		apt-get install -y python3-distutils
+	fi
+}
+
+# Gets the name of the python executable for python starting with python3, then
+# checking python. Sets the variable to an empty string if neither are found.
+get_python_path() {
+	python_path=""
+	if [ -f /bin/python3 ]; then
+		python_path="/bin/python3"
+	elif [ -f /bin/python ]; then
+		python_path="/bin/python"
+	else
+		python_path=$(which python3 2>/dev/null)
```

Comment:
nit: what if only python2 is in the path?

Reply:
Thanks for catching this, changed this to an elif. The variable should only be set if python3 is found.

Reply:
and what if python is in the path? should you not register it?

Reply:
Updated to only use which, moving away from hard coded paths.